lib/puppet/provider/kernel_parameter/grub2.rb

```diff
@@ -35,19 +35,12 @@ def self.mkconfig_path
   #
   # @return (String) The commandline
   def self.mkconfig_cmdline
-    needs_bls_cmdline = lambda do
-      os = Facter.value(:os)
-      return false unless os.is_a?(Hash)
-
-      return false unless os['family'] == 'RedHat'
-      return false if os['release']['major'].to_i < 9
-      return false if os['release']['major'].to_i == 9 && os['release']['minor'].to_i < 3
-
-      true
-    end
+    os = Facter.value(:os)
+    needs_bls_cmdline = os.is_a?(Hash) && os['family'] == 'RedHat' && !%w[Amazon Fedora].include?(os['name']) &&
+                        os['release']['major'].to_i == 9 && os['release']['minor'].to_i >= 3
```

Comment:
When will Amazon or Fedora have `major` 9? Fedora 9 is long gone and Amazon did 1, 2, 2023…
So wouldn't this be the same and easier to read?
```suggestion
    needs_bls_cmdline = os.is_a?(Hash) && os['family'] == 'RedHat' && os['release']['major'].to_i == 9 && os['release']['minor'].to_i >= 3
```

Reply:
I suggested that in slack, but it does feel "hacky" this makes the intent obvious

Reply:
Would a comment make it also more obvious?
While I agree this is very explicit, it's hard to read (IMHO).

Reply:
I am fine either way, it was suggested by @ekohl , so I changed it the way he requested :)

Reply:
My biggest issue was with the many `return false` statements. To me that was very confusing.

I did discuss the simplification in https://github.com/voxpupuli/puppet-augeasproviders_grub/pull/103#pullrequestreview-2248357529